### PR TITLE
Improve solidity tests

### PR DESCRIPTION
**Pull request by seanyoung**
Signed-off-by: Sean Young <sean@mess.org>

### CODE REVIEW

**Comment by seanyoung** [comment]
https://github.com/paritytech/scripts/pull/202 is needed for this to work (requires newer Solang).

File: tests/contracts-solidity.spec.ts [hunk 1]
```diff
@@ -63,12 +66,7 @@ describe("Solang Smart Contracts", () => {
   test("Raw Flipper contract", async (done): Promise<void> => {
     // The next two lines are a not so pretty workaround until the new metadata format has been fully implemented
     const metadata = require("../contracts/solidity/flipper/flipper.json");
-    const selector = u8aToHex(new Uint8Array(
-      JSON.parse(metadata.contract.constructors[0].selector)
-    ));
-    const flipAction = u8aToHex(new Uint8Array(
-     JSON.parse(metadata.contract.messages[0].selector)
-    ));
+    const abi = new Abi(api.registry, metadata);
```
Comment (Stefie) [inline tests/contracts-solidity.spec.ts:69]: Ah, cool, that's working again? It used to be broken so I haven't tried it in a while!

Comment (seanyoung) [reply tests/contracts-solidity.spec.ts:69]: It seems to be working. I have no idea how to decode return values though.

File: tests/utils.ts [hunk 2]
```diff
@@ -90,6 +90,25 @@ export async function callContract(
   await sendAndReturnFinalized(signer, tx);
 }
 
+export async function rpcContract(
+  api: ApiPromise,
+  dest: Address,
+  inputData: any,
+  gasLimit: number = 2147483648,
```
Comment (Stefie) [inline tests/utils.ts]: Could you use `GAS_REQUIRED` here. The naming is not quite correct anymore after some changes in the contracts pallet. You can also rename it to `GAS_LIMIT`, if that feels better.

File: tests/utils.ts [hunk 3]
```diff
@@ -90,6 +90,25 @@ export async function callContract(
   await sendAndReturnFinalized(signer, tx);
 }
 
+export async function rpcContract(
+  api: ApiPromise,
+  dest: Address,
```
Comment (Stefie) [inline tests/utils.ts]: It's called `contractAddress` in the other util functions, I'd prefer to keep the same name in all of them. 
I named it `contractAddress` to make it more obvious for external people looking at the code what we're referring to, but I'm also fine with using the substrate prop name `dest`. Could you change it in the other functions, too, then?

File: tests/contracts-solidity.spec.ts [hunk 4]
```diff
@@ -98,18 +96,90 @@ describe("Solang Smart Contracts", () => {
...
     expect(flipBack.toString()).toEqual("0x01");
 
+    res = await rpcContract(api, address, abi.messages.get());
+    expect(res.toString()).toEqual("0x01");
+
+    done();
+  });
+
+  test("Raw Creator contract", async (done): Promise<void> => {
+    // The next two lines are a not so pretty workaround until the new metadata format has been fully implemented
```
Comment (Stefie) [inline tests/contracts-solidity.spec.ts]: If `const abi = new Abi()` is working now, this line can be deleted :)

File: tests/utils.ts (hunk 2 above)
Comment (seanyoung) [reply tests/utils.ts]: GAS_REQUIRED does not work, I get a message saying it does not fit in a u32. I'll create a GAS_LIMIT.

File: tests/utils.ts (hunk 3 above)
Comment (seanyoung) [reply tests/utils.ts]: Renamed to contractAddress

File: tests/contracts-solidity.spec.ts (hunk 4 above)
Comment (seanyoung) [reply tests/contracts-solidity.spec.ts]: Removed

File: tests/utils.ts (hunk 2 above)
Comment (Stefie) [reply tests/utils.ts]: Interesting, I thought it changed to u64 altogether...

Comment (seanyoung) [reply tests/utils.ts]: That would make sense, however current substrate is not accepting it. I've just pulled latest master in substrate and says:
 `   2020-06-02 17:09:27        RPC-CORE: call(callRequest: ContractCallRequest, at?: BlockHash): ContractExecResult:: -32602: `100000000000` > u32::max_value(), the max block number is u32.`
Note the block number. This probably a bug in @polkadot/api-contract.

**Pull request merged**